common/src/main/java/com/github/twitch4j/common/util/TwitchLimitRegistry.java

```diff
@@ -0,0 +1,101 @@
+package com.github.twitch4j.common.util;
+
+import com.github.twitch4j.common.enums.TwitchLimitType;
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Bucket;
+import io.github.bucket4j.Bucket4j;
+import io.github.bucket4j.BucketConfiguration;
+import io.github.bucket4j.IncompatibleConfigurationException;
+import io.github.bucket4j.local.LocalBucketBuilder;
+import lombok.NonNull;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This singleton facilitates sharing of key rate-limit buckets by user and limit type.
+ * <p>
+ * For example:
+ * <pre>
+ * {@code TwitchLimitRegistry.getInstance().setLimit(
+ *         "149223493",
+ *         TwitchLimitType.CHAT_MESSAGE_LIMIT,
+ *         Collections.singletonList(TwitchChatLimitHelper.MOD_MESSAGE_LIMIT)
+ *     );
+ * }
+ * </pre>
+ */
+public enum TwitchLimitRegistry {
+
+    /**
+     * The single thread-safe instance of the limit registry.
+     */
+    INSTANCE;
+
+    private final Map<String, Map<TwitchLimitType, Bucket>> limits = new ConcurrentHashMap<>();
+
+    /**
+     * Attempts to set a user's bucket for a specific rate-limit.
+     *
+     * @param userId    the id of the user whose rate limit is being specified.
+     * @param limitType the type of rate-limit that is being configured.
+     * @param limit     the bandwidths that are applicable for this user and rate limit type.
+     * @throws IncompatibleConfigurationException if overriding an old limit that is incompatible with the new configuration
+     */
+    public void setLimit(@NonNull String userId, @NonNull TwitchLimitType limitType, @NonNull List<Bandwidth> limit) {
```

Comment:
Should there be versions where `List<Bandwidth>` is replaced with `Bandwidth[]` or `Bandwidth` (or is it fine to make the user call `Arrays.asList` or `Collections.singletonList` respectively)

Reply:
I think it's fine, there is no need to overload this method.